disk_store.go

```diff
@@ -147,8 +153,13 @@ func (d *DiskStore) Set(key string, value string) error {
 	// 1. Encode the KV into bytes
 	// 2. Write the bytes to disk by appending to the file
 	// 3. Update KeyDir with the KeyEntry of this key
+
+	if err := validateKV(key, []byte(value)); err != nil {
+		return err
+	}
+
 	timestamp := uint32(time.Now().Unix())
-	h := Header{TimeStamp: timestamp, KeySize: uint32(len(key)), ValueSize: uint32(len(value))}
+	h := Header{CheckSum: crc32.ChecksumIEEE([]byte(value)), TimeStamp: timestamp, KeySize: uint32(len(key)), ValueSize: uint32(len(value))}
```

Comment:
checksum should be calculated on all the values. What if my KeySize / Key is corrupted?